src/scenarios-general-helpers.mod.js

```diff
@@ -1,107 +1,151 @@
+  log.debug('Input config path: "{}"', configPath);
   var config = readConfig(configPath);
 
   if (!config) {
-    log.error("Error: Could not read config from " + configPath);
+    log.error('Error: Could not read config from "{}"', configPath);
     return null;
   }
-  log.debug("The input config contains: " + JSON.stringify(config));
+  log.debug('The input config contains: "{}"', JSON.stringify(config));
 
   if (!config.hasOwnProperty('configVersion')) {
-    log.error("Error: 'configVersion' does not exist in the configuration.");
+    log.error('"configVersion" does not exist in the configuration');
     return null;
   }
-  log.debug("'configVersion' exist in the configuration.");
 
   if (config.configVersion !== reqGeneralCfgVer) {
-    log.error("Global config version mismatch. Expected version: " + reqGeneralCfgVer + ", but got: " + config.configVersion);
+    log.error(
+      'Global config version mismatch. Expected version: "{}", but got: "{}"',
+      reqGeneralCfgVer,
+      config.configVersion
+    );
     return null;
   }
-  log.debug("Global config version is valid.");
 
-  // Проверяем существование поля, тип массив, что не пуст
   if (!config.hasOwnProperty('scenarios')) {
-    log.error("Error: 'scenarios' does not exist in the configuration.");
+    log.error('"scenarios" does not exist in the configuration');
     return null;
   }
 
   var listAllScenarios = config.scenarios;
   if (!Array.isArray(listAllScenarios)) {
-    log.error("Error: 'scenarios' is not an array.");
+    log.error('"scenarios" is not an array');
     return null;
   }
 
   if (listAllScenarios.length === 0) {
-    log.debug("'scenarios' array is empty.");
+    log.debug('"scenarios" array is empty');
     return null;
   }
 
-  log.debug("Config 'scenarios' array is correct.");
+  log.debug('Config "scenarios" array is correct');
   return listAllScenarios;
 }
 
-exports.findAllActiveScenariosWithType = function (listScenario,
-                                                   searchScenarioType,
-                                                   reqScenarioCfgVer) {
-  var res = findAllActiveScenariosWithType(listScenario,
-                                           searchScenarioType,
-                                           reqScenarioCfgVer);
+/**
+ * Returns the ID prefix based on the provided configuration or
+ *     transliterates the device title
+ * @param {string} deviceTitle The device title used for transliteration
+ *     if 'idPrefix' is not provided
+ * @param {Object} cfg The configuration object containing
+ *     the 'idPrefix' property
+ * @return {string} The ID prefix
+ */
+function getIdPrefix(deviceTitle, cfg) {
+  var idPrefix = '';
+  var isIdPrefixProvided = cfg.idPrefix && cfg.idPrefix.trim() !== '';
+
+  if (isIdPrefixProvided === true) {
+    idPrefix = cfg.idPrefix;
+  } else {
+    idPrefix = translit(deviceTitle);
+  }
+  return idPrefix;
+}
+
+exports.findAllActiveScenariosWithType = function (
```

Comment:
```suggestion
exports.findAllActiveScenariosWithType = findAllActiveScenariosWithType;
```
минус 11 строк кода

И так же везде ниже

Reply:
Спасибо

раньше пробовал без этой конструкции экспортировать из модуля функции, но почему то результат в этом случае не возвращался из такой экспортированной функции

Сейчас проверил, вроде работает - странно

Стало так
```
exports.findAllActiveScenariosWithType = findAllActiveScenariosWithType;
exports.readAndValidateScenariosConfig = readAndValidateScenariosConfig;
exports.getIdPrefix = getIdPrefix;
```